core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/RecordingTaskManagerListener.java

```diff
@@ -49,14 +49,14 @@ public RecordingTaskManagerListener(DiscoveryNode localNode, String... actionMas
     @Override
     public synchronized void onTaskRegistered(Task task) {
         if (Regex.simpleMatch(actionMasks, task.getAction())) {
-            events.add(new Tuple<>(true, task.taskInfo(localNode, true)));
+            events.add(new Tuple<>(true, task.taskInfo(localNode, System.currentTimeMillis(), true)));
```

Comment:
If you make current time a settable thing then you can have exact control over the run time during you tests I think. Useful? You'd have to have task creation delegate to the same current time implementation but I think that isn't too hard?